sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -165,6 +165,24 @@ object ResolveHints {
           hintErrorHandler.hintRelationsNotFound(h.name, h.parameters, unmatchedIdents)
           applied
         }
+      case With(child, relations) => resolveCTEHint(child,
```

Comment:
This patch looks like it is specifically designed to fix `With` comparing to #29062. I am open to more comments.